subsys/bluetooth/host/scan.h

```diff
@@ -5,12 +5,95 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#ifndef SUBSYS_BLUETOOTH_HOST_SCAN_H_
+#define SUBSYS_BLUETOOTH_HOST_SCAN_H_
+
+#include <stdint.h>
+
+#include <zephyr/sys/atomic.h>
+#include <zephyr/bluetooth/bluetooth.h>
+
+/**
+ * Reasons why a scanner can be running.
+ * Used as input to @ref bt_le_scan_user_add
+ * and @ref bt_le_scan_user_remove.
+ */
+enum bt_le_scan_enabled_flag {
+	/** The application explicitly instructed the stack to scan for advertisers
+	 * using the API @ref bt_le_scan_start().
+	 * May not be used explicitly.
+	 */
+	BT_LE_EXPLICIT_SCAN,
+
+	/**
+	 * Syncing to a periodic advertiser.
+	 */
+	BT_LE_SYNC_SYNCING,
+
+	/**
+	 * Scanning to find devices to connect to.
+	 */
+	BT_LE_SCAN_BEFORE_INITIATE,
+
+	/**
+	 * Special state for a NOP for @ref bt_le_scan_update_and_reconfigure to not add/remove any
+	 * flag why the scanner should be running.
+	 */
+	BT_LE_NONE,
+	BT_LE_NUM_FLAGS,
+};
```

Comment:
@jori-nordic you requested the names of these to be shorter. I'm not against that, but I do feel like these are too generic.e. `BT_LE_NONE` in itself contains litte to no context. 

What do you think?

Reply:
I'm sorry I got misunderstood. GH is not a nice review tool. 

This was just a suggestion in order to fit <100chars for that one problematic line. The nak was on other comments. I agree that `BT_LE_NONE` is a bit non-sensical. I'm ok with reverting to what you suggested before @Thalley

Reply:
Based on the current enum, I'd rename it to `bt_le_scan_flag` and use that as a prefix for all values, and have `bt_le_scan_flag_disable` instead of `none` (or omit both, as the omission of any of the others implies `none`/`disabled`)

Reply:
What about BT_LE_SCAN_USER_* ?

Reply:
that's actually better yeah